wa/commands/process.py

```diff
@@ -0,0 +1,116 @@
+
+class ProcessContext(object):
+
+    def __init__(self):
+        self.run_output = None
+        self.target_info = None
+        self.job_output = None
+
+class ProcessCommand(Command):
+
+    name = 'process'
+    description = 'Process the output from previously run workloads.'
+
+    def initialize(self, context):
+        self.parser.add_argument('directory', metavar='DIR',
+                                 help="""
+                                 Specify a directory containing the data
+                                 from a previous run to be processed.
+                                 """)
+        self.parser.add_argument('-p', '--processor', action='append',
+                                 dest='additional_processors', metavar='OutputProcessor',
+                                 help="""
+                                 Specify an output processor to add from the
+                                 command line. This can be used to run a
+                                 processor that is not normally used without
+                                 introducing permanent change to the config 
+                                 (which one might then forget to revert). This
+                                 option may be specified multiple times.
+                                 """)
+        self.parser.add_argument('-f', '--force', action='store_true',
+                                 help="""
+                                 Run processors that have already been
+                                 run. By default these will be skipped.
+                                 """)
+        self.parser.add_argument('-r', '--recursive', action='store_true',
+                                 help="""
+                                 Walk the specified directory to process
+                                 all of the previous runs contained within
+                                 instead of just processing the root.
+                                 """)
+
+    def execute(self, config, args):
+        process_directory = os.path.expandvars(args.directory)
+        self.logger.debug('Using process directory: {}'.format(process_directory))
+        if not os.path.exists(process_directory):
+            msg = 'Path `{}` does not exist, please specify a valid path.'
+            raise CommandError(msg.format(process_directory))
+        if not args.recursive:
+            output_list = [RunOutput(process_directory)]
+        else:
+            output_list = [output for output in discover_wa_outputs(process_directory)]
+        pc = ProcessContext()
+        for run_output in output_list:
+            if not args.recursive:
+                self.logger.info('Installing output processors')
+            else:
+                self.logger.info('Install output processors for run in path `{}`'
+                                 .format(run_output.basepath))
+            pm = ProcessorManager()
```

Comment:
Should instantiate with `loader=config.plugin_cache` here.

Reply:
Hm, OK, will do. I thought I had tried it and it failed on the install so I reverted but very preliminary testing just now suggests otherwise.

Reply:
Yes, it fails if you specify a processor on the command line that's not in the config because there's no `get_output_processor` function. I can try to work around that if you like?

Reply:
Should be fixed by https://github.com/ARM-software/workload-automation/pull/713